frontend/app/src/store/balances/getters.ts

```diff
@@ -947,29 +948,113 @@ export const getters: Getters<
       const asset = assetInfo(identifier);
       return asset?.symbol ?? identifier;
     },
+  locationBreakdown:
+    (
+      {
+        connectedExchanges,
+        manualBalances,
+        loopringBalances,
+        prices
+      }: BalanceState,
+      { exchangeBalances, totals },
+      { session }
+    ) =>
+    identifier => {
+      const ignoredAssets = session!.ignoredAssets;
+      const ownedAssets: { [asset: string]: AssetBalanceWithPrice } = {};
+      const addToOwned = (value: AssetBalance) => {
+        const asset = ownedAssets[value.asset];
+        if (ignoredAssets.includes(value.asset)) {
+          return;
+        }
+        ownedAssets[value.asset] = !asset
+          ? {
+              ...value,
+              usdPrice: prices[value.asset] ?? new BigNumber(-1)
+            }
+          : {
+              asset: asset.asset,
+              amount: asset.amount.plus(value.amount),
+              usdValue: asset.usdValue.plus(value.usdValue),
+              usdPrice: prices[asset.asset] ?? new BigNumber(-1)
+            };
+      };
+
+      const exchange = connectedExchanges.find(
+        ({ location }) => identifier === location
+      );
+
+      if (exchange) {
+        const balances = exchangeBalances(identifier);
+        balances.forEach((value: AssetBalance) => addToOwned(value));
+      }
+
+      if (identifier === TRADE_LOCATION_BLOCKCHAIN) {
+        totals.forEach((value: AssetBalance) => addToOwned(value));
+
+        for (const address in loopringBalances) {
+          const balances = loopringBalances[address];
+          for (const asset in balances) {
+            addToOwned({
+              ...balances[asset],
+              asset
+            });
+          }
+        }
+      }
+
+      manualBalances.forEach(value => {
+        if (value.location === identifier) {
+          addToOwned(value);
+        }
+      });
+
+      return Object.values(ownedAssets).sort((a, b) =>
+        b.usdValue.minus(a.usdValue).toNumber()
+      );
+    },
   byLocation: (
     state,
-    { blockchainTotal, exchanges, manualBalanceByLocation: manual }
+    {
+      blockchainTotal,
+      exchangeRate,
+      exchanges,
+      manualBalanceByLocation: manual
+    },
+    { session }
   ) => {
     const byLocation: BalanceByLocation = {};
 
     for (const { location, usdValue } of manual) {
       byLocation[location] = usdValue;
     }
 
+    const mainCurrency = session?.generalSettings.mainCurrency.tickerSymbol;
+    assert(mainCurrency, 'main currency was not properly set');
+
+    const currentExchangeRate = exchangeRate(mainCurrency);
+    const blockchainTotalConverted = currentExchangeRate
+      ? blockchainTotal.multipliedBy(currentExchangeRate)
+      : blockchainTotal;
```

Comment:
So this check will work when there is no exchange rate for the selected currency, but which cases are these specifically? ETH/BTC?

Reply:
I'm not sure though. I saw that the currentExchangeRate can be `undefined` at `manualBalanceByLocation` getter, so I thought it is common to check it.

Reply:
no keep it, but I don't remember why the check is on manual balance, the only point I can think of this possibly being undefined is during the dashboard loading, but I am under the impression that we load the exchange rates early during the login.